### PR TITLE
Regenerate Using Latest Swagger to Add Encryption Features

### LINKED ISSUE #15908: Regenerate Search libraries from swagger
With https://github.com/Azure/azure-rest-api-specs/pull/10839 having been merged for api-version 2020-06-30, please regenerate models. This PR adds `encryptionKey` to `searchIndexer`, `searchIndexerDataSource`, and `searchIndexerSkillsets` but should look similar to `searchIndex.encryptionKey` which uses the `searchResourceEncryptionKey` model.

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #15908

This PR regenerates `azure-search-documents` to add encryption features that were recently added into the swagger.

### CODE REVIEW

**Comment by alzimmermsft** [comment]
@heaths is the new SHA being used what you were planning on targeting?

**Comment by heaths** [comment]
I wasn't yet targeting anything specific. Latetly, I've been taking the SHA of the last commit that touched any files we care about, so if this is that SHA, great. But even the latest master commit SHA would be fine. The idea is that regenerating doesn't yield any surprises down the road.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/SearchIndexerSkillsetConverter.java [hunk 1]
```diff
@@ -69,8 +71,13 @@ public static SearchIndexerSkillset map(com.azure.search.documents.indexes.imple
         String description = obj.getDescription();
         searchIndexerSkillset.setDescription(description);
 
-        String eTag = obj.getETag();
-        searchIndexerSkillset.setETag(eTag);
+        searchIndexerSkillset.setDescription(obj.getDescription());
```
Comment (mssfang) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/SearchIndexerSkillsetConverter.java:74]: does these setDescription() call twice?

**Review by mssfang — APPROVED** [review]
one comment and LGTM!

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/SearchIndexerSkillsetConverter.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/SearchIndexerSkillsetConverter.java:74]: Two different methods, one converts from the codegen type to hand-written and the other from hand-written to codegen.

**Pull request merged**